torch/ao/quantization/fx/prepare.py

```diff
@@ -821,11 +824,26 @@ def propagate_dtypes_for_known_nodes(
     replace this with a better way to reason about dtypes of tensors.
     """
     for node in graph.nodes:
-        bool_arg_idxs = node_bool_tensor_arg_indexes(node)
-        for bool_arg_idx in bool_arg_idxs:
-            cur_node = node.args[bool_arg_idx]
-            maybe_propagate_dtype_for_node(
-                cur_node, torch.bool, node_name_to_target_dtype, matches)
+        non_observable_arg_dict = get_non_observable_arg_indexes_and_types(node)
+
+        for arg_type in non_observable_arg_dict:
+            non_observable_indices = non_observable_arg_dict[arg_type](node)
+
+            for index in non_observable_indices:
+                arg = node.args[index]
+
+                # when an argument is a tuple, it does not show up as another node so we need to go through
+                # all elements of the tuple manually
+                if isinstance(arg, tuple):
+                    arg_list = list(arg)
+                else:
+                    arg_list = [arg]
```

Comment:
`arg` could be a nested tuple/list/dict right?

Reply:
I didn't realize it could be a dict, I'll add a test case for this and make sure to handle it

Reply:
as far as I can tell it looks like it turns every key/value pair in a dict into its own node. I added a test case where I made a module that accepts a dict as an argument to test this though